addons/mass_mailing/static/src/builder/plugins/social_media_option_plugin.js

```diff
@@ -0,0 +1,429 @@
+            editingElement.append(element);
+        }
+        this.dependencies.history.addStep();
+    }
+
+    /** @returns {Array[string]} */
+    get platforms() {
+        return [
+            "social_twitter",
+            "social_facebook",
+            "social_github",
+            "social_linkedin",
+            "social_youtube",
+            "social_instagram",
+            "social_tiktok",
+            "social_discord",
+        ];
+    }
+
+    get baseMediaValues() {
+        return {
+            facebook: "https://www.facebook.com/Odoo",
+            instagram: "https://www.instagram.com/explore/tags/odoo/",
+            twitter: "https://x.com/Odoo",
+            linkedin: "https://www.linkedin.com/company/odoo",
+            tiktok: "https://www.tiktok.com/@odoo",
+        };
+    }
+
+    /** @param {integer} companyId */
+    async fetchRecordedSocialMedia(companyId) {
+        if (!companyId) {
+            return {};
+        }
+        const records = await this.services.orm.read("res.company", [companyId], this.platforms);
+        const medias = {};
+        for (const [fieldName, href] of Object.entries(records[0])) {
+            if (!href) {
+                continue;
+            }
+            const matches = fieldName.match(/social_(\w+)/);
+            if (matches) {
+                const platform = matches[0];
+                if (this.platforms.includes(platform)) {
+                    medias[platform] = href;
+                }
+            }
+        }
+        this.medias = medias;
+        return medias;
+    }
+
+    addMedia(platform, href) {
+        this.medias[platform] = href;
+    }
+
+    getMedias() {
+        return { ...this.medias };
+    }
+
+    removeMedia(platform) {
+        delete this.medias[platform];
+    }
+}
+
+class UpdateSocialMediaLinksAction extends BuilderAction {
+    static id = "updateSocialMediaLinks";
+    static dependencies = ["massMailingSocialMediaOptionPlugin"];
+    apply({ editingElement, params: { mainParam: platform }, value }) {
+        const linkToEdit = editingElement.querySelector(`a[data-platform='${platform}']`);
+        if (!linkToEdit) {
+            return;
+        }
+        linkToEdit.setAttribute("href", value);
+    }
+    getValue({ editingElement, params: { mainParam: platform } }) {
+        const linkToEdit = editingElement.querySelector(`a[data-platform='${platform}']`);
+        if (!linkToEdit) {
+            return this.dependencies.massMailingSocialMediaOptionPlugin.getMedias()[platform] ?? "";
+        }
+        return linkToEdit.getAttribute("href");
+    }
+}
+
+class ChangeCompanyAction extends BuilderAction {
+    static id = "changeCompany";
+    static dependencies = ["massMailingSocialMediaOptionPlugin"];
+    async apply({ editingElement, params: company }) {
+        if (this.isApplied(...arguments)) {
+            return;
+        }
+        editingElement.querySelector(".o_social_snippet_dropped_placeholder")?.remove();
```

Comment:
is it possible to change company while a snippet is being dropped? Because this placeholder is only supposed to be there while the snippet is being dropped, doesn'it? I'm not sure it is necessary to remove that element here, but maybe you had an issue at some point?